pkg/provisioners/channel_util.go

```diff
@@ -179,20 +221,28 @@ func newK8sService(c *eventingv1alpha1.Channel) *corev1.Service {
 	}
 }
 
+func k8sServiceLabels(c *eventingv1alpha1.Channel) map[string]string {
+	return map[string]string{
+		"channel":     c.Name,
```

Comment:
The downside of this change is that all the existing `Service`s and `VirtualService`s would immediately be ignored. New generated name versions would be created immediately (although the existing ones would continue to exist until the `Channel` was deleted). 

Hmm, maybe that isn't so bad. Calling the old `Service` should still work. Calling the new one would work. 

One thing I would need to verify is what happens if there are multiple `VirtualService`s that attempt to rewrite the same host, as both the old and new `VirtualService` would attempt to rewrite `channel-name.channel-namespace.channels.cluster.local`. They would both try to rewrite it in the same way, but I don't know if multiple definitions would be problematic. 

I'll test out what happens with the multiple `VirtualService`s. Assuming that works, I'll make the change. Are these the labels you want?

```
eventing.knative.dev/channel: c.Name
eventing.knative.dev/provisioner: c.Spec.Provisioner.Name
```

Reply:
Yes. One additional option would be to keep them both for one release?

Reply:
It looks like it is unsupported to have multiple `VirtualService`s use the same `host` with a 'catch-all' rule. Also, the sidecar doesn't seem to support multiple `VirtualService`s with the same `host`. From the [docs](https://istio.io/help/ops/traffic-management/deploy-guidelines/#multiple-virtual-services-and-destination-rules-for-the-same-host):

> A VirtualService can only be fragmented this way if it is bound to a gateway. Host merging is not supported in sidecars.

In fact, I can see the following logs in a sidecar when I tried to make the change:

> [2018-12-10 21:26:47.448][21][warning][config] bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_mux_subscription_lib/common/config/grpc_mux_subscription_impl.h:70] gRPC config for type.googleapis.com/envoy.api.v2.RouteConfiguration rejected: Only unique values for domains are permitted. Duplicate entry of domain gcp.default.channels.cluster.local

So, if we change the labels, things may happen to work, but Istio could choose to break that compatibility at any time. How they choose to break compatibility is undefined, and may result in non-working `Channel`s (e.g. they reject the entire `VirtualService` rather than just the problematic portion). If we still choose to change the labels, then I would suggest we tell users to either manually bounce all `Channel`s or to manually delete old `VirtualServices`.

We can make this backwards compatible. In this release, search based on the old labels and have the controller add the both the new and old labels to all `Service`s and `VirtualService`s. Next release, only recognize the new labels (and stop writing them to existing `Service`s and `VirtualService`s). This means that anyone upgrading 0.2 -> 0.3 -> 0.4 works. 0.2 -> 0.4 would have the same possible problems discussed above.

Do you still want to change them? If so, breaking or backwards compatible?